src/test/analyzer_test.ts

```diff
@@ -523,4 +549,193 @@ suite('Analyzer', () => {
           testSeed.events.map(e => e.name), ['fired-event', 'data-changed']);
     });
   });
+
+  suite('race conditions and caching', () => {
+    test.skip('editor simulator of imports that import a common dep', async() => {
+      // Here we're simulating a lot of noop-changes to base.html, which has
+      // two imports, which mutually import a common dep. This stresses the
+      // analyzer's caching.
+
+      const wait = () =>
+          new Promise((resolve) => setTimeout(resolve, Math.random() * 30));
```

Comment:
I refactored this code to log the wait times that are used in the case that it fails, and added a skipped test that uses such an array of wait times. Sadly, this didn't seem to reliably reproduce the same failure, it looks like, between the setTimeout scheduler's degrees of freedom and other system factors there's still enough nondeterminism to keep the failures from being deterministic. 

Still, it's better than nothing and may be useful the next time we're called back into this code, so I'll leave it for now and let us decide then whether it should be kept.